collaboration/extending-kyma-cli.md

```diff
@@ -0,0 +1,346 @@
+
+##### Expanded command with switches
+
+`kyma function create <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function update <function-name> -n <namespace> --runtime <runtime> --min-mem <mb> --max-mem <mb> --min-cpu <cpu> --max-cpu <cpu> --min-replicas 1 --max-replicas 1 --src-code <path-to-src-code> --dependecies <path-to-dep-file>`
+
+`kyma function delete <function-name> -n <namespace>`
+
+`kyma function get <function-name> -n <namespace>`
+
+*Labels*
+
+`kyma function label <function-name> ["foo=bar",..] -n <namespace>`
+
+*Expose*
+
+`kyma function expose <function-name> --secure(optional) --actions <GET/POST> -n <namespace>`
+
+*Service Catalog*
+
+`kyma function bind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-instaces [<binding-instance-name>,...] -n <namespace>`
+
+`kyma function bind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+`kyma function unbind <function-name> --binding-usage [<binding-usage-name>,...] -n <namespace>`
+
+*Bind to events*
+
+`kyma function trigger <function-name> --events [<event-name>,..] -n <namespace>`
+`kyma function trigger <function-name> --events [<event-name>,..] -n <namespace>`
+
+*Bind to environment variables*
+
+`kyma function bind <function-name> --env [<foo=bar>,..] -n <namespace>`
+`kyma function unbind <function-name> --env [<foo=bar>,..] -n <namespace>`
+
+*List all functions*
+
+`kyma function list -n <namespace>`
+
+> A detailed table view with function name, runtime, age, replicas and state
+
+*Runtimes*
+
+`kyma function get runtimes`
+
+*Logs*
+
+`kyma function logs <function-name> -n <namespace>`
+
+*function Status*
+
+`kyma function status <function-name> -n <namespace>`
+
+Should display status
+
+```bash
+  Name: <function-name>
+  Runtime: <my-runtime>
+  State: Running/Error/Deploying/...
+  Replicas: 1
+  Min-replicas: 1
+  Max-replicas: 2
+  API: <url>
+    Secure: yes/no
+    Actions: GET, POST
+  Bindings:
+    Events: <list of events>
+    Servicebindings: <list of service bindings>
+  labels: foo:bar
+```
+
+`kyma function show <function-name> events`
+`kyma function show <function-name> apis`
+`kyma function show <function-name> service-binding-usages`
+`kyma function show <function-name> labels`
+`kyma function show <function-name> replicas`
+
+#### Typical User flow
+
+This section details a typical development flow for the functions
+
+1. New User creating function from scratch or from existing github project as shown [here](#creation-of-project-locally). This would generate a function project for vscode. User is also provided with readme file to explain the structure of project and also the config.yaml. It should also explain how user can test it locally and also debugging.
```

Comment:
Would work in general with git I guess:

```suggestion
1. New User creating function from scratch or from existing git project as shown [here](#creation-of-project-locally). This would generate a function project for vscode. User is also provided with readme file to explain the structure of project and also the config.yaml. It should also explain how user can test it locally and also debugging.
```

Reply:
removed this.